qlib/linux_def.rs

```diff
@@ -3017,13 +3019,16 @@ impl MemoryDef {
 }
 
 #[cfg(target_arch = "aarch64")]
-impl MemoryDef{
-    // use 2 pages below PHY_LOWER_ADDR as MMIO base.
-    pub const HYPERCALL_MMIO_BASE: u64 = Self::KVM_IOEVENTFD_BASEADDR - Self::PAGE_SIZE;
-    pub const HYPERCALL_MMIO_SIZE: u64 = 0x1000;
+impl MemoryDef {
+    pub const USER_UPPER_ADDR: u64 = Self::HYPERCALL_MMIO_BASE;
+    //
+    // One (device) page, placed two pages bellow the PHY_LOWER_ADDR.
+    //
+    pub const HYPERCALL_MMIO_BASE: u64 = Self::KVM_IOEVENTFD_BASEADDR
+                                         - 2 * Self::PAGE_SIZE;
```

Comment:
Why minus two pages here?
(KVM_IOEVENTFD_BASEADDR is already one page below PHY_LOWER_ADDR, IIRC)

Reply:
btw, just keep in mind the IOEVENTFD is obsolete and we will remove it later.  (don't worry about it for now).